### PR TITLE
Using Poetry for dependency management and adjustments for docker

### PR DESCRIPTION
**Pull request by SavinaRoja**
https://github.com/mouse-reeve/bookwyrm/issues/793

Brought the branch up to date. Give this a good review. I think I did not fully update `bw-dev`. It looks like it is used basically like a Makefile?

### CODE REVIEW

**Comment by mouse-reeve** [comment]
Yeah, I'm not familiar with using Makefiles so maybe that would have been a more sensible or common way to approach it, but bw-dev is just meant to be helpful shortcut commands for common administrative tasks.

**Comment by SavinaRoja** [comment]
Whatever you like works.

Just for the sake of example, here is a snip from one of my projects which is pretty straightforward:

```Makefile
black:
        poetry run black .

flake8:
        poetry run flake8

bandit:
        poetry run bandit -r project

isort:
        poetry run isort .

mypy:
        poetry run mypy

lint: isort black flake8 bandit mypy

test:
        poetry run python -m pytest tests

run-flask:
        poetry run flask run

run-command-worker:
        poetry run celery worker -A run_project_worker.celery -Q celery -n="command_worker@project"

run-periodic-worker:
        poetry run celery worker -A run_project_worker.celery -B -Q data_integrity -n="periodic@project"

run-flower:
        poetry run flower -A run_project_worker.celery

start:
        docker-compose up

install: build

build:
        docker-compose build
```

So when developing, one can install the full suite of dependencies using `poetry install`, then you could run a full lint process using `make lint` or you could run tests with `make test`. It also has commands for if you want to run the components outside of docker.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,28 @@
-FROM python:3.9
+FROM python:3.9.2-slim-buster
```
Comment (SavinaRoja) [inline Dockerfile:1]: This seems to be working for me without problem, and should give a smaller image. But I feel like this is a bit out of scope and I can take this out.

**Comment by DeeUnderscore** [comment]
Hey, I wanted to note that I've been using this, with some changes, to [deploy BookWyrm on NixOS](https://github.com/DeeUnderscore/bookwyrm/tree/nix/nix) (without Docker), since it works well with poetry2nix. Notable change which would probably be useful in Docker deployment, too, is adding Gunicorn, which is how BookWyrm runs in the production branch.

**Comment by mouse-reeve** [comment]
Since everyone who's voiced an opinion seems happy about using Poetry, I'm happy to merge when this is ready

**Comment by mouse-reeve** [comment]
I'm going through and closing pull requests that are stalled or out of date -- if you'd like to get this PR back in action, I'm super happy to re-open and work on getting it in sync with the current state of `main` ❤️

**Pull request closed without merging**